src/cdk-experimental/menu/menu.md

```diff
@@ -0,0 +1,241 @@
+When a user right-clicks within the context the associated menu opens up next to their cursor - as
+you would expect with a typical right click menu.
+
+<!-- TODO basic context menu example -->
+
+Nested components can have context menu triggers at each level. When a user right clicks, the menu
+associated with the deepest component the user is hovering over is opened up.
+
+```html
+<div [cdkContextMenuTriggerFor]="outer">
+  My outer context
+  <div [cdkContextMenuTriggerFor]="inner">
+    My inner context
+  </div>
+</div>
+```
+
+In the example above, right clicking on "My inner context" will open up the "inner" menu and right
+clicking inside "My outer context" will open up the "outer" menu.
+
+### Inline Menus
+
+Inline menus are groups of menu items which typically do not trigger their own menus and the menu
+items implement some type of on-click behavior. Unlike triggered menus which are opened up from some
+menu item, inline menus do not pop up - they are always visible. The benefit of using an inline menu
+over a group of standalone menu items is that inline menus provide keyboard navigation,
+accessibility and focus management. Menu items within an inline menus are logically grouped
+together.
+
+<!-- TODO inline menu example (gmail buttons?) -->
+
+### Menu Items
+
+Components which interact with or are placed inside of a menu or menubar should typically have the
+`cdkMenuItem` directive provided. This directive allows the items to be navigated to via keyboard
+interaction.
+
+A menuitem by itself can provide some user defined action by hooking into the `cdkMenuItemTriggered`
+output. An example may be a close button which performs some closing logic.
+
+```html
+<ng-template cdkMenuPanel #panel="cdkMenuPanel">
+  <div cdkMenu [cdkMenuPanel]="panel">
+    <button cdkMenuItem (cdkMenuItemTriggered)="closeApp()">Close</button>
+  </div>
+</ng-template>
+```
+
+`cdkMenuItem` should also be added to a component which triggers a menu or submenu.
+
+```html
+<ng-template cdkMenuPanel #panel="cdkMenuPanel">
+  <div cdkMenu [cdkMenuPanel]="panel">
+    <button cdkMenuItem [cdkMenuTriggerFor]="submenu">Open Submenu</button>
+  </div>
+</ng-template>
+```
+
+A menuitem also has two sub-types, neither of which should trigger a menu: CdkMenuItemCheckbox and
+CdkMenuItemRadio
+
+#### Menu Item Checkboxes
+
+A component with the `cdkMenuItemCheckbox` directive behaves just as you would expect a typical
+checkbox to behave. A use case may be toggling an independent setting since it contains a checked
+state. A component with the `cdkMenuItemCheckbox` directive does not need the additional
+`cdkMenuItem` directive.
+
+#### Menu Item Radios
+
+A `cdkMenuItemRadio` behaves as you would expect a radio button to behave. A use case may be
+toggling through more than one state since radio buttons can be grouped together. A component with
+the `cdkMenuItemRadio` directive does not need the additional `cdkMenuItem` directive.
+
```

Comment:
```suggestion
A `cdkMenuItemRadio` is a special type of menuitem that behaves as a radio button. You
can use this type of menuitem for menus with exclusively selectable items.

```

Reply:
I think leaving `A component with the
`cdkMenuItemRadio` directive does not need the additional `cdkMenuItem` directive.` is worth it. Thoughts?

Here and for checkbox